src/modules/NeighborInfoModule.cpp

```diff
@@ -106,12 +106,13 @@ void NeighborInfoModule::sendNeighborInfo(NodeNum dest, bool wantReplies)
     meshtastic_NeighborInfo neighborInfo = meshtastic_NeighborInfo_init_zero;
     collectNeighborInfo(&neighborInfo);
     meshtastic_MeshPacket *p = allocDataProtobuf(neighborInfo);
-    // send regardless of whether or not we have neighbors in our DB,
-    // because we want to get neighbors for the next cycle
     p->to = dest;
     p->decoded.want_response = wantReplies;
     p->priority = meshtastic_MeshPacket_Priority_BACKGROUND;
-    printNeighborInfo("SENDING", &neighborInfo);
+    // only send neighbours if we have some to send
+    if (neighborInfo.neighbors_count > 0) {
+        printNeighborInfo("SENDING", &neighborInfo);
+    }
     service->sendToMesh(p, RX_SRC_LOCAL, true);
```

Comment:
`sendToMesh()` should also be in the if-clause then, but better to check it before allocating the data protobuf to avoid memory leak also.

Reply:
D'oh!